src/Hyperlink/index.tsx

```diff
@@ -1,29 +1,45 @@
 import React from 'react';
 import PropTypes from 'prop-types';
 import classNames from 'classnames';
-import isRequiredIf from 'react-proptype-conditional-require';
 import { Launch } from '../../icons';
 import Icon from '../Icon';
 
-import withDeprecatedProps, { DeprTypes } from '../withDeprecatedProps';
-
 export const HYPER_LINK_EXTERNAL_LINK_ALT_TEXT = 'in a new tab';
 export const HYPER_LINK_EXTERNAL_LINK_TITLE = 'Opens in a new tab';
 
-const Hyperlink = React.forwardRef((props, ref) => {
-  const {
-    className,
-    destination,
-    children,
-    target,
-    onClick,
-    externalLinkAlternativeText,
-    externalLinkTitle,
-    variant,
-    isInline,
-    showLaunchIcon,
-    ...attrs
-  } = props;
+interface Props extends Omit<React.ComponentPropsWithoutRef<'a'>, 'href' | 'target'> {
+  /** specifies the URL */
+  destination: string;
+  /** Content of the hyperlink */
+  children: React.ReactNode;
+  /** Custom class names for the hyperlink */
+  className?: string;
+  /** Alt text for the icon indicating that this link opens in a new tab, if target="_blank". e.g. _("in a new tab") */
+  externalLinkAlternativeText?: string;
```

Comment:
I left this as an optional prop. There was propTypes code below to make `externalLinkAlternativeText` required when `target="_blank"` but this wasn't actually working as far as I can tell, especially since a default value is given.

If we want to make those props conditionally required (when target=`_blank`), I _can_ express that in the TypeScript types, but there are several instances in this repo alone that do not do that ([example](https://github.com/openedx/paragon/blob/9993fb3b938e50012b5aa220c1e75f4b8ff684c6/www/src/components/Menu.tsx#L218-L221)) and I suspect it would then show type errors in many other MFEs as well (if MFEs even use this component? Maybe they use react-router's Link instead).

We probably _should_ make that required at some point, because the default value is not internationalized. But the only good way to do that that I could think of is to require each MFE to subclass `Hyperlink` with a component that provides internationalized defaults for these fields, and that seems like too much work. Is there not some way to provide internationalized defaults in Paragon itself?

Reply:
> but this wasn't actually working as far as I can tell, especially since a default value is given.

Yeah, you're likely correct the `ifRequiredWhen` isn't really doing much due to the default value.

> I suspect it would then show type errors in many other MFEs as well (if MFEs even use this component? Maybe they use react-router's Link instead).
Yes, this is usually the case. See the recently filed issue https://github.com/openedx/paragon/issues/3050 regarding adapting `Hyperlink` to be compatible with `Link`, such that `Link` usage can have more consistent styling.

> We probably should make that required at some point, because the default value (HYPER_LINK_EXTERNAL_LINK_ALT_TEXT, above) is not internationalized. But the only good way to do that that I could think of is to require each MFE to subclass Hyperlink with a component that provides internationalized defaults for these fields, and that seems like too much work. Is there not some way to provide internationalized defaults in Paragon itself?

Paragon has had some i18n support in its components for a couple years now (see [ADR](https://github.com/openedx/paragon/blob/master/docs/decisions/0015-i18n-support.rst) and the [README](https://github.com/openedx/paragon/tree/9993fb3b938e50012b5aa220c1e75f4b8ff684c6?tab=readme-ov-file#internationalization)), except IIRC the `pull_translations` job routinely/currently fails due to an translation format error of some kind that's prevented the translations from updating in awhile. Related, I'm also not sure if Paragon has been migrated to using `openedx-atlas` / `openedx-translations` yet (unlikely), or if it intends to; @brian-smith-tcril might be able to share some insight about any plans to migrate.

It does seem by moving translations to `openedx-translations`, most MFEs are no longer importing/supplying `paragonMessages` anymore (e.g., removed from frontend-app-learning in [this commit](https://github.com/openedx/frontend-app-learning/commit/94d20833aaa635e17f3ff953a268553dab3638ef); I'm personally not sure if the Paragon i18n messages (and the shared header/footer component i18n messages) are somehow still passed to the MFEs, or if that support was (intentionally?) removed as part of the migration to `openedx-translations`?

If we want it to have a translated default prop (I agree, we should; looks like `Hyperlink` was missed in the original introduction of i18n for hardcoded English strings), then ideally we mark the strings for i18n via react-intl's `defineMessage` / `intl.formatMessage` or `FormattedMessage`, similar to a handful of other components in Paragon today; if the prop is provided, it gets used (where consumer might pass its own i18n string), otherwise falling back to the default i18n Paragon message.

Reply:
Thanks for the context on i18n! I'm glad to see there is something in place, at least partially. I can look at fixing that in a follow-up PR.

Also great to see there's a plan to support `Link` :)